# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../libsmbios/gcc43.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- libsmbios-0.13.2/bins/smitest.cpp.vanilla	2008-04-09 14:22:29.000000000 +0000
+++ libsmbios-0.13.2/bins/smitest.cpp	2008-04-09 14:22:41.000000000 +0000
@@ -22,6 +22,7 @@
 #include <string>
 #include <iostream>
 #include <iomanip>
+#include <string.h>
 #include <stdlib.h>
 
 #include "smbios/ISmi.h"  // only needed if you want to use fake input (memdump.dat)
--- libsmbios-0.13.2/bins/ascii2enUS_scancode.cpp.vanilla	2008-04-09 14:22:50.000000000 +0000
+++ libsmbios-0.13.2/bins/ascii2enUS_scancode.cpp	2008-04-09 14:22:59.000000000 +0000
@@ -20,6 +20,7 @@
 #include "smbios/compat.h"
 
 #include <iostream>
+#include <string.h>
 
 #include "smbios/ISmi.h"
 #include "smbios/version.h"
--- libsmbios-0.13.2/libraries/smi/Smi.cpp.vanilla	2008-04-09 14:12:53.000000000 +0000
+++ libsmbios-0.13.2/libraries/smi/Smi.cpp	2008-04-09 14:13:05.000000000 +0000
@@ -20,7 +20,7 @@
 #define LIBSMBIOS_SOURCE
 #include "smbios/compat.h"
 
-//#include <iostream>
+#include <string.h>
 
 #include "SmiImpl.h"
 #include "smbios/ISmbios.h"
--- libsmbios-0.13.2/libraries/smi/Smi_Linux.cpp.vanilla	2008-04-09 14:14:21.000000000 +0000
+++ libsmbios-0.13.2/libraries/smi/Smi_Linux.cpp	2008-04-09 14:14:30.000000000 +0000
@@ -21,6 +21,7 @@
 #include "smbios/compat.h"
 
 #include <sys/file.h>
+#include <string.h>
 
 #include "SmiImpl.h"
 
--- libsmbios-0.13.2/libraries/token/TokenD4.cpp.vanilla	2008-04-09 14:16:44.000000000 +0000
+++ libsmbios-0.13.2/libraries/token/TokenD4.cpp	2008-04-09 14:16:52.000000000 +0000
@@ -20,6 +20,7 @@
 #define LIBSMBIOS_SOURCE
 #include "smbios/compat.h"
 
+#include <string.h>
 #include <iomanip>
 
 #include "TokenImpl.h"
--- libsmbios-0.13.2/libraries/token/TokenDA.cpp.vanilla	2008-04-09 14:17:38.000000000 +0000
+++ libsmbios-0.13.2/libraries/token/TokenDA.cpp	2008-04-09 14:17:49.000000000 +0000
@@ -20,6 +20,7 @@
 #define LIBSMBIOS_SOURCE
 #include "smbios/compat.h"
 
+#include <string.h>
 #include <iomanip>
 
 #include "TokenImpl.h"
--- libsmbios-0.13.2/libraries/token/TokenD5.cpp.vanilla	2008-04-09 14:17:00.000000000 +0000
+++ libsmbios-0.13.2/libraries/token/TokenD5.cpp	2008-04-09 14:17:07.000000000 +0000
@@ -20,6 +20,7 @@
 #define LIBSMBIOS_SOURCE
 #include "smbios/compat.h"
 
+#include <string.h>
 #include <sstream>
 #include <iomanip>
 
--- libsmbios-0.13.2/libraries/token/TokenD6.cpp.vanilla	2008-04-09 14:17:21.000000000 +0000
+++ libsmbios-0.13.2/libraries/token/TokenD6.cpp	2008-04-09 14:17:28.000000000 +0000
@@ -20,6 +20,7 @@
 #define LIBSMBIOS_SOURCE
 #include "smbios/compat.h"
 
+#include <string.h>
 #include <sstream>
 #include <iomanip>
 
--- libsmbios-0.13.2/libraries/systeminfo/System.cpp.vanilla	2008-04-09 14:15:18.000000000 +0000
+++ libsmbios-0.13.2/libraries/systeminfo/System.cpp	2008-04-09 14:15:27.000000000 +0000
@@ -34,6 +34,8 @@
 // this always should be included last
 #include "smbios/message.h"
 
+#include <string.h>
+
 using namespace smbios;
 using namespace cmos;
 using namespace std;
--- libsmbios-0.13.2/libraries/systeminfo/SystemDetect.cpp.vanilla	2008-04-09 14:16:14.000000000 +0000
+++ libsmbios-0.13.2/libraries/systeminfo/SystemDetect.cpp	2008-04-09 14:16:27.000000000 +0000
@@ -33,6 +33,8 @@
 // include this last.
 #include "smbios/message.h"
 
+#include <string.h>
+
 using namespace smbios;
 using namespace cmos;
 using namespace std;
--- libsmbios-0.13.2/libraries/systeminfo/IdByte.cpp.vanilla	2008-04-09 14:14:43.000000000 +0000
+++ libsmbios-0.13.2/libraries/systeminfo/IdByte.cpp	2008-04-09 14:15:06.000000000 +0000
@@ -31,6 +31,8 @@
 // should always be included last.
 #include "smbios/message.h"
 
+#include <string.h>
+
 using namespace smbios;
 using namespace cmos;
 using namespace std;
--- libsmbios-0.13.2/libraries/xml_libxml2/SmbiosXml.cpp.vanilla	2008-04-09 14:18:13.000000000 +0000
+++ libsmbios-0.13.2/libraries/xml_libxml2/SmbiosXml.cpp	2008-04-09 14:18:20.000000000 +0000
@@ -20,6 +20,7 @@
 #define LIBSMBIOS_SOURCE
 #include "smbios/compat.h"
 
+#include <string.h>
 #include <sstream>
 #include <iomanip>
 
--- libsmbios-0.13.2/libraries/memory/Memory_Linux.cpp.vanilla	2008-04-09 14:09:53.000000000 +0000
+++ libsmbios-0.13.2/libraries/memory/Memory_Linux.cpp	2008-04-09 14:10:14.000000000 +0000
@@ -23,6 +23,7 @@
 #include <errno.h>
 #include <sys/mman.h>   /* mmap */
 #include <unistd.h>     /* getpagesize */
+#include <string.h>	/* strerror */
 
 #include "MemoryImpl.h"
 
--- libsmbios-0.13.2/libraries/memory/Memory.cpp.vanilla	2008-04-09 14:10:26.000000000 +0000
+++ libsmbios-0.13.2/libraries/memory/Memory.cpp	2008-04-09 14:10:35.000000000 +0000
@@ -22,6 +22,7 @@
 
 #include <errno.h>
 #include <stdio.h>
+#include <string.h>
 
 #include "MemoryImpl.h"
 
--- libsmbios-0.13.2/libraries/smbios/SmbiosStrategy.cpp.vanilla	2008-04-09 14:11:02.000000000 +0000
+++ libsmbios-0.13.2/libraries/smbios/SmbiosStrategy.cpp	2008-04-09 14:11:10.000000000 +0000
@@ -20,6 +20,7 @@
 #define LIBSMBIOS_SOURCE
 #include "smbios/compat.h"
 
+#include <string.h>
 #include <sstream>
 
 #include "smbios/IMemory.h"
--- libsmbios-0.13.2/libraries/smbios/SmbiosWorkaround.cpp.vanilla	2008-04-09 14:12:12.000000000 +0000
+++ libsmbios-0.13.2/libraries/smbios/SmbiosWorkaround.cpp	2008-04-09 14:12:35.000000000 +0000
@@ -26,6 +26,8 @@
 // message.h should be included last.
 #include "smbios/message.h"
 
+#include <string.h>
+
 using namespace std;
 
 // convenience function.
--- libsmbios-0.13.2/libraries/smbios/SmbiosTable.cpp.vanilla	2008-04-09 14:11:40.000000000 +0000
+++ libsmbios-0.13.2/libraries/smbios/SmbiosTable.cpp	2008-04-09 14:11:48.000000000 +0000
@@ -20,6 +20,7 @@
 #define LIBSMBIOS_SOURCE
 #include "smbios/compat.h"
 
+#include <string.h>
 #include <sstream>
 
 #include "SmbiosImpl.h"
--- libsmbios-0.13.2/libraries/smbios/SmbiosItem.cpp.vanilla	2008-04-09 14:11:19.000000000 +0000
+++ libsmbios-0.13.2/libraries/smbios/SmbiosItem.cpp	2008-04-09 14:11:30.000000000 +0000
@@ -20,6 +20,7 @@
 #define LIBSMBIOS_SOURCE
 #include "smbios/compat.h"
 
+#include <string.h>
 #include <iomanip>
 
 #include "SmbiosImpl.h"
--- libsmbios-0.13.2/supported-bins/dellBiosUpdate.cpp.vanilla	2008-04-09 14:22:04.000000000 +0000
+++ libsmbios-0.13.2/supported-bins/dellBiosUpdate.cpp	2008-04-09 14:22:11.000000000 +0000
@@ -20,6 +20,7 @@
 #include "smbios/compat.h"
 
 #include <iostream>
+#include <string.h>
 
 #include "smbios/DellRbu.h"
 #include "smbios/SystemInfo.h"
--- libsmbios-0.13.2/supported-bins/verifySmiPassword.cpp.vanilla	2008-04-09 14:20:41.000000000 +0000
+++ libsmbios-0.13.2/supported-bins/verifySmiPassword.cpp	2008-04-09 14:20:49.000000000 +0000
@@ -19,6 +19,7 @@
 // compat header should always be first header if including system headers
 #include "smbios/compat.h"
 
+#include <string.h>
 #include <iostream>
 
 #include "smbios/ISmi.h"
--- libsmbios-0.13.2/supported-bins/serviceTag.cpp.vanilla	2008-04-09 14:20:05.000000000 +0000
+++ libsmbios-0.13.2/supported-bins/serviceTag.cpp	2008-04-09 14:20:13.000000000 +0000
@@ -22,6 +22,7 @@
 #include <string>
 #include <iostream>
 #include <iomanip>
+#include <string.h>
 #include <stdlib.h>
 
 #include "smbios/IMemory.h"  // only needed if you want to use fake input (memdump.dat)
--- libsmbios-0.13.2/supported-bins/dellLcdBrightness.cpp.vanilla	2008-04-09 14:21:30.000000000 +0000
+++ libsmbios-0.13.2/supported-bins/dellLcdBrightness.cpp	2008-04-09 14:21:56.000000000 +0000
@@ -22,6 +22,7 @@
 #include <string>
 #include <iostream>
 #include <iomanip>
+#include <string.h>
 #include <stdlib.h>
 
 #include "smbios/ISmi.h"
--- libsmbios-0.13.2/supported-bins/assetTag.cpp.vanilla	2008-04-09 14:18:48.000000000 +0000
+++ libsmbios-0.13.2/supported-bins/assetTag.cpp	2008-04-09 14:19:00.000000000 +0000
@@ -22,6 +22,7 @@
 #include <string>
 #include <iostream>
 #include <iomanip>
+#include <string.h>
 #include <stdlib.h>
 
 #include "smbios/IMemory.h"  // only needed if you want to use fake input (memdump.dat)
--- libsmbios-0.13.2/supported-bins/propertyTag.cpp.vanilla	2008-04-09 14:19:07.000000000 +0000
+++ libsmbios-0.13.2/supported-bins/propertyTag.cpp	2008-04-09 14:19:40.000000000 +0000
@@ -22,6 +22,7 @@
 #include <string>
 #include <iostream>
 #include <iomanip>
+#include <string.h>
 #include <stdlib.h>
 
 #include "smbios/ISmi.h"  // only needed if you want to use fake input (memdump.dat)
--- libsmbios-0.13.2/supported-bins/tokenCtl.cpp.vanilla	2008-04-09 14:21:01.000000000 +0000
+++ libsmbios-0.13.2/supported-bins/tokenCtl.cpp	2008-04-09 14:21:19.000000000 +0000
@@ -35,6 +35,7 @@
 #include <string>
 #include <iostream>
 #include <iomanip>
+#include <string.h>
 #include <stdlib.h>
 
 #include "smbios/IToken.h"
--- libsmbios-0.13.2/include/smbios/ISmi.h.vanilla	2008-04-09 14:13:28.000000000 +0000
+++ libsmbios-0.13.2/include/smbios/ISmi.h	2008-04-09 14:13:50.000000000 +0000
@@ -31,6 +31,8 @@
 // abi_prefix should be last header included before declarations
 #include "smbios/config/abi_prefix.hpp"
 
+#include <memory>
+
 namespace smi
 {
     //! Abstract base class for the smi read write operations
